### PR TITLE
online-judge-tools: init at 11.5.1

### PR DESCRIPTION
**Pull request by sei40kr**
###### Motivation for this change

This PR adds [online-judge-tools](https://github.com/online-judge-tools/oj).

One of its dependency `online-judge-api-client` clones/updates `library-checker-problems` by internally invoking `git` command, and then run `generate.py` in the repository.
~So I added those dependencies to the `PATH/PYTHONPATH`.~
So I created another Python environment to run `generate.py`.

https://github.com/online-judge-tools/api-client/blob/master/onlinejudge/service/library_checker.py#L48-L70
https://github.com/online-judge-tools/api-client/blob/master/onlinejudge/service/library_checker.py#L93-L111
https://github.com/yosupo06/library-checker-problems/blob/master/requirements.txt

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/online-judge-api-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ buildPythonPackage, fetchPypi, git, lib, python3Packages }:
```
Comment (jonringer) [inline pkgs/development/python-modules/online-judge-api-client/default.nix:1]: Using `pythonXPackages` isn't allowed from `python-modules` as this will mix interpreter versions for a given package. For example, if you were to do `python37Packages.<mypkg>`, but the expression pulled from `python3[8]Packages`. Python will be mis-importing several libraries, and this will likely cause issues at runtime.

Please refer to [Python dependency handling](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies) on the best practices for handling dependencies within `python-modules`. Namely, you should be importing each python package individually, and not refer to another python package set (E.g. `python3Packages`).

If you're new to nixpkgs, you're free to view some of my videos on nixpkgs and python:
- [Python packaging using nixpkgs](https://www.youtube.com/watch?v=jXd-hkP4xnU)
- [Adding a package to nixpkgs unstable](https://www.youtube.com/watch?v=fvj8H5yUKu8)

File: pkgs/development/python-modules/online-judge-api-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  version = "10.10.0";
+  format = "wheel";
+
+  src = fetchPypi {
+    inherit version format;
+    pname = "online_judge_api_client";
+    dist = "py3";
+    python = "py3";
+    sha256 = "932dafa971d0cd3ebdb7444e049dc0719f87b93123af020e4d3d1bbe499e9142";
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/online-judge-api-client/default.nix:14]: I would really prefer to use the source over the wheel: https://github.com/online-judge-tools/oj

File: pkgs/tools/misc/online-judge-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{ git, lib, python3Packages, ... }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "online-judge-tools";
+  version = "11.5.1";
+  format = "wheel";
+
```
Comment (jonringer) [inline pkgs/tools/misc/online-judge-tools/default.nix:7]: if this is going to be packaged as a module. then all you need to do is `  online-judge-tools = with python3Packages; toPythonApplication online-judge-tools;` in `all-packages.nix`

File: pkgs/tools/misc/online-judge-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  preFixup = ''
+    makeWrapperArgs=(
+      --prefix PATH : ${lib.makeBinPath [ git ]}
+      --prefix PYTHONPATH : ${
+        with python3Packages;
+        python3Packages.makePythonPath [ colorlog jinja2 markdown toml ]
+      }
+    )
+  '';
```
Comment (jonringer) [inline pkgs/tools/misc/online-judge-tools/default.nix:33]:
```suggestion
    makeWrapperArgs = [
      "--prefix PATH : ${lib.makeBinPath [ git ]}"
    ];
```

The python path is already exported by `buildPythonXXX`

File: pkgs/tools/misc/online-judge-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    colorama
+    requests
+  ];
+
+  # Wheels have no tests
+  doCheck = false;
+
+  preFixup = ''
+    makeWrapperArgs=(
+      --prefix PATH : ${lib.makeBinPath [ git ]}
```
Comment (jonringer) [inline pkgs/tools/misc/online-judge-tools/default.nix:27]: The most proper fix would be to patch the source code, so that it works even when being used as a module

File: pkgs/tools/misc/online-judge-tools/default.nix (hunk 4 above)
Comment (sei40kr) [reply pkgs/tools/misc/online-judge-tools/default.nix:33]: @jonringer 
These Python modules is not for online-judge-tools, but for [library-checker-problems](https://github.com/yosupo06/library-checker-problems) that online-judge-tools clones and run internally.

First I tried all these dependencies to `propagatedBuildInputs`, but it didn't work.
Is there a smarter way to solve this?

Comment (jonringer) [reply pkgs/tools/misc/online-judge-tools/default.nix:33]: `propagatedBuildInputs` should work

Comment (sei40kr) [reply pkgs/tools/misc/online-judge-tools/default.nix:33]: @jonringer I tried again, and it didn't work.

```nix
  propagatedBuildInputs =
    [ colorama colorlog jinja2 markdown online-judge-api-client requests toml ];
```

```
$ ./result/bin/oj dl https://judge.yosupo.jp/problem/predecessor_problem
[INFO] online-judge-tools 11.5.1 (+ online-judge-api-client 10.10.0)
[INFO] load cookie from: /home/sei40kr/.local/share/online-judge-tools/cookie.jar
git version 2.33.1
[INFO] $ /nix/store/zhf7xxp7k1605wirn29kylg1x2bl4pq3-git-2.33.1/bin/git -C /home/sei40kr/.cache/online-judge-tools/library-checker-problems pull
Already up to date.
[INFO] $ /nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9 /home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py /home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml
Traceback (most recent call last):
  File "/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py", line 20, in <module>
    import toml
ModuleNotFoundError: No module named 'toml'
[ERROR] the generate.py failed: check https://github.com/yosupo06/library-checker-problems/issues
[ERROR] Command '['/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/main.py", line 101, in main
    sys.exit(run_program(parsed, parser=parser))
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/main.py", line 61, in run_program
    if not subcommand_download.run(args):
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/subcommand/download.py", line 120, in run
    samples = problem.download_sample_cases(session=sess)
  File "/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages/onlinejudge/service/library_checker.py", line 78, in download_sample_cases
    self._generate_test_cases_in_cloned_repository()
  File "/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages/onlinejudge/service/library_checker.py", line 108, in _generate_test_cases_in_cloned_repository
    subprocess.check_call(command, stdout=sys.stderr, stderr=sys.stderr)
  File "/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml']' returned non-zero exit status 1.
```

Comment (sei40kr) [reply pkgs/tools/misc/online-judge-tools/default.nix:33]: @jonringer 
It seems `propagatedBuildInputs` are appended to `sys.path` by `site.addsitedir`.

```python
import sys;import site;import functools;sys.argv[0] = '/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/bin/oj';functools.reduce(lambda k, p: site.addsitedir(p, k), ['/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages','/nix/store/wark7kwccm3kn06h5xwhhvv5rcg8331i-python3.9-colorama-0.4.4/lib/python3.9/site-packages','/nix/store/wvncjga0w2hvsnszwa7j16p57wdkgh0y-python3.9-colorlog-6.5.0/lib/python3.9/site-packages','/nix/store/jpdalbpwrqp3gsn1cnjzivmsr390s982-python3.9-Jinja2-3.0.2/lib/python3.9/site-packages','/nix/store/z84rmnhy1grx4jls23n5aay25rp29aad-python3.9-Babel-2.9.1/lib/python3.9/site-packages','/nix/store/rj4cwhhkr8hm2w1xxw26b5kmka9541hn-python3.9-pytz-2021.3/lib/python3.9/site-packages','/nix/store/hfvwfl2pvlj5x2dm85vgfgmkll627dgd-python3.9-markupsafe-2.0.1/lib/python3.9/site-packages','/nix/store/qsqgwd1y8m47vlch7qw6qpx90zwg90ql-python3.9-Markdown-3.3.4/lib/python3.9/site-packages','/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages','/nix/store/0fijgd5j5g6l4gxkdlmpbs8dq1s8lg3f-python3.9-appdirs-1.4.4/lib/python3.9/site-packages','/nix/store/r3fh82iy0ialr2rivl86c6hw6ppah4gn-python3.9-beautifulsoup4-4.10.0/lib/python3.9/site-packages','/nix/store/arhdy6kri2yrb22w2an6lf0f59zhcn70-python3.9-html5lib-1.1/lib/python3.9/site-packages','/nix/store/7ixnxb28f4sg2c08a1pnp3yxdb0h48cs-python3.9-six-1.16.0/lib/python3.9/site-packages','/nix/store/bwyfa04hx5rm9k01vg13iizhs2gzdbnq-python3.9-webencodings-0.5.1/lib/python3.9/site-packages','/nix/store/2xwdhgbr58zx3b394acj4wa82bxv2v1y-python3.9-lxml-4.6.3/lib/python3.9/site-packages','/nix/store/dgnzpgfnkm78kv24r4r02ylk4lzjpbsa-python3.9-soupsieve-2.2.1/lib/python3.9/site-packages','/nix/store/jy7nn3w05g60ns5vrlw3ywjnkd5vpyx1-python3.9-jsonschema-3.2.0/lib/python3.9/site-packages','/nix/store/n8bc0fxv5sp10s9xrpr210dzj5pv5x7i-python3.9-attrs-21.2.0/lib/python3.9/site-packages','/nix/store/j192qx3qyr5w8050qq9s4p001k7d6w52-python3.9-importlib-metadata-4.8.1/lib/python3.9/site-packages','/nix/store/2vhiiavpj8d98x9ja3c80ra985wpx6g6-python3.9-toml-0.10.2/lib/python3.9/site-packages','/nix/store/kwkj2bk4zd08sd7j5qrhkxfkjk4cif45-python3.9-zipp-3.6.0/lib/python3.9/site-packages','/nix/store/lscg5l570grdka03jymahqhgcdnp0c35-python3.9-more-itertools-8.10.0/lib/python3.9/site-packages','/nix/store/b7w6b69hgcxfmhsm4hq9xwygbym0hc37-python3.9-pyrsistent-0.18.0/lib/python3.9/site-packages','/nix/store/wnd6rnxc2hqsam2r280xv4yy1bygd08n-python3.9-requests-2.26.0/lib/python3.9/site-packages','/nix/store/q6zv2g1pyjfzxphgsvr0qbm2xbfvx7jq-python3.9-certifi-2021.10.08/lib/python3.9/site-packages','/nix/store/bx6dm7hzd6dnbaifj0ydhiwxfqf90125-python3.9-idna-3.2/lib/python3.9/site-packages','/nix/store/lg6hps3px9isdhcm5jhv4x3c3xwpz3i7-python3.9-urllib3-1.26.7/lib/python3.9/site-packages','/nix/store/c8izknabvsnbrmqahmpd5pgdm0wz1pdy-python3.9-brotli-1.0.9/lib/python3.9/site-packages','/nix/store/z262iy04g0lj2g4w5gqks3i7n4dm28fc-python3.9-pysocks-1.7.1/lib/python3.9/site-packages','/nix/store/ichac696i1qr8dxvmhipnxp1af4cdfgi-python3.9-chardet-4.0.0/lib/python3.9/site-packages','/nix/store/kzb17mnvgci0mhs3pa8hdga67f8nmqxj-python3.9-brotlicffi-1.0.9.2/lib/python3.9/site-packages','/nix/store/4flpmcvw0p795bx1s9lnam7kyl7qhviv-python3.9-pycparser-2.20/lib/python3.9/site-packages','/nix/store/wf8vlq8gibp76vp816yw0fxjlc5ldphs-python3.9-cffi-1.14.6/lib/python3.9/site-packages','/nix/store/hmci2h8g3rydi7qrc9h2nmyvw9kff8l9-python3.9-charset-normalizer-2.0.5/lib/python3.9/site-packages'], site._init_pathinfo());
```

But `sys.path` is not exported to a subprocess, so I think we need another fix for this.

**Comment by sei40kr** [comment]
@jonringer 
I fixed the code where your commented. Could you review again?

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -33694,4 +33694,6 @@ with pkgs;
   };
 
   zthrottle = callPackage ../tools/misc/zthrottle { };
+
+  online-judge-tools = with python3.pkgs; toPythonApplication online-judge-tools;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33698]: Please sort this entry into a fitting category.

File: pkgs/development/python-modules/online-judge-tools/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  propagatedBuildInputs = [ colorama online-judge-api-client requests ];
+
+  # Requires internet access
+  doCheck = false;
+
+  meta = with lib; {
+    description =
+      "Tools for various online judges. Downloading sample cases, generating additional test cases, testing your code, and submitting it.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/default.nix:22]:
```suggestion
    description = "Tools for various online judges. Download sample cases, generat additional test cases, test your code, and submit it.";
```
I am not sure what is proper English grammar but I think before it was a bit off.

File: pkgs/development/python-modules/online-judge-tools/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "online-judge-tools";
+    repo = "oj";
+    rev = "v${version}";
+    sha256 = "0zkzmmjgjb6lyrzq1ip54cpnp7al9a7mcyjyi5vx58bvnx3q0c6m";
+  };
+
+  propagatedBuildInputs = [ colorama online-judge-api-client requests ];
+
+  # Requires internet access
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/default.nix:18]: Please add a pythonImportsCheck.

File: pkgs/development/python-modules/online-judge-tools/default.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
+{ buildPythonPackage, colorama, fetchFromGitHub, lib, online-judge-api-client
+, requests, ... }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/default.nix:2]:
```suggestion
{ buildPythonPackage, colorama, fetchFromGitHub, lib, online-judge-api-client, requests }:
```

File: pkgs/development/python-modules/online-judge-tools/api-client.nix [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      src = ./api-client-fix-library-checker-problems-generate-py-deps.patch;
+      pythonInterpreter = pythonEnv.interpreter;
+    })
+  ];
+
+  propagatedBuildInputs =
+    [ appdirs beautifulsoup4 colorlog jsonschema lxml requests toml ];
+
+  # Requires internet access
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/api-client.nix:33]: Please add a pythonImportsCheck.

File: pkgs/development/python-modules/online-judge-tools/api-client.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      inherit git;
+    })
+    (substituteAll {
+      src = ./api-client-fix-library-checker-problems-generate-py-deps.patch;
+      pythonInterpreter = pythonEnv.interpreter;
+    })
+  ];
+
+  propagatedBuildInputs =
+    [ appdirs beautifulsoup4 colorlog jsonschema lxml requests toml ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/api-client.nix:30]:
```suggestion
  propagatedBuildInputs = [ appdirs beautifulsoup4 colorlog jsonschema lxml requests toml ];
```

File: pkgs/development/python-modules/online-judge-tools/api-client.nix [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+  patches = [
+    (substituteAll {
+      src = ./api-client-fix-paths.patch;
+      inherit git;
+    })
+    (substituteAll {
+      src = ./api-client-fix-library-checker-problems-generate-py-deps.patch;
+      pythonInterpreter = pythonEnv.interpreter;
+    })
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/api-client.nix:26]: This could also be done via substituteInPlace which maybe breaks less often. Either is fine.

File: pkgs/development/python-modules/online-judge-tools/api-client.nix [hunk 13]
```diff
@@ -0,0 +1,41 @@
+{ appdirs, beautifulsoup4, buildPythonPackage, colorlog, fetchFromGitHub, git
+, jsonschema, lib, lxml, markdown, python, requests, substituteAll, toml }:
+
+let
+  pythonEnv =
+    python.withPackages (ps: with ps; [ colorlog jinja2 markdown toml ]);
+in buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-tools/api-client.nix:7]:
```suggestion
buildPythonPackage rec {
```

I am not sure if we are allowed to do this in pythonPackages but my gut feeling says no.

Comment (sei40kr) [reply pkgs/development/python-modules/online-judge-tools/api-client.nix:7]: As we discussed https://github.com/NixOS/nixpkgs/pull/146267#discussion_r750499985, this is necessary for online-judge-tools to internally download [library-checker-problems](https://github.com/yosupo06/library-checker-problems) and invoke `generate.py` in it.

Is there a better way?
I already tried `propagatedBuildInputs` and it didn't work.

Comment (sei40kr) [reply pkgs/development/python-modules/online-judge-tools/api-client.nix:7]: I considered I should make a derivation for library-checker-problems and contain it in online-judge-tools.
But library-checker-problems is more contents than a tool, and is often updated. So I don't want to pin its revision.

File: pkgs/top-level/all-packages.nix [hunk 14]
```diff
@@ -34375,4 +34375,6 @@ with pkgs;
   honeytail = callPackage ../servers/tracing/honeycomb/honeytail { };
 
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
+
+  online-judge-tools = with python3.pkgs; toPythonApplication online-judge-tools;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34379]: Please sort this into a fitting category and sort there alphabetically.

File: pkgs/development/python-modules/online-judge-tools/api-client.nix (hunk 13 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/online-judge-tools/api-client.nix:7]: If nothing of this is propagated this should not be a problem. Please add a comment that other people know that.

**Comment by sei40kr** [comment]
@SuperSandro2000 
I've fixed all the parts where you reviewed.

Could you review again, please? Thank you.

File: pkgs/development/python-modules/online-judge-api-client/default.nix [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    beautifulsoup4
+    colorlog
+    jsonschema
+    lxml
+    requests
+    toml
+  ];
+
+  # Requires internet access
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/online-judge-api-client/default.nix:50]:
```suggestion
  doCheck = false;

  pythonImportsCheck = [ "onlinejudge" "onlinejudge_api" ];
```

File: pkgs/development/python-modules/online-judge-tools/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/online-judge-tools/default.nix:23]:
```suggestion
  doCheck = false;

  pythonImportsCheck = [ "onlinejudge_command" ];
```

File: pkgs/development/python-modules/online-judge-api-client/default.nix (hunk 15 above)
Comment (sei40kr) [reply pkgs/development/python-modules/online-judge-api-client/default.nix:50]: @SuperSandro2000 Fixed.

File: pkgs/development/python-modules/online-judge-tools/default.nix [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+, buildPythonPackage
+, colorama
+, fetchFromGitHub
+, online-judge-api-client
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "online-judge-tools";
+  version = "11.5.1";
```
Comment (jonringer) [inline pkgs/development/python-modules/online-judge-tools/default.nix:11]: this should probably be a separate commit, but this PR is ancient, and don't want to cause yet-another-turnaround

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 146267 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  <ul>    <li>online-judge-tools (python39Packages.online-judge-tools)</li>    <li>python310Packages.online-judge-api-client</li>    <li>python310Packages.online-judge-tools</li>    <li>python39Packages.online-judge-api-client</li>  </ul></details>
